components/sync/driver/brave_sync_profile_sync_service.cc

```diff
@@ -140,6 +144,13 @@ void BraveProfileSyncService::ReenableSyncTypes() {
 
 bool BraveProfileSyncService::IsReenableTypesRequired(
     const SyncCycleSnapshot& snapshot) {
+  if (!last_reenable_types_time_.is_null() &&
+      base::Time::Now() - last_reenable_types_time_ <
+          kMinimalTimeBetweenReenable) {
+    // Can't do reenable more often than once in a specified amount
```

Comment:
I think we should also do a P3A for this case so we can observe abnormality and revisit why it happens so frequently. And just add it to `Brave.Sync.TypesEverReenabled`, making it buckets of 2, one for "normal reenabling types due to 7th failure", another one for "reenabling types aborted due to 7th failure happening twice in a row in less than 30mins"

Reply:
modified at https://github.com/brave/brave-core/pull/7805/commits/cb4972500305b4f83b2a477e33e929a19339f8ea